scriptmodules/supplementary/splashscreen.sh

```diff
@@ -27,22 +27,39 @@ function _video_exts_splashscreen() {
 }
 
 function depends_splashscreen() {
-    getDepends fbi omxplayer
+    getDepends fbi omxplayer insserv
 }
 
 function install_bin_splashscreen() {
-    cp "$md_data/asplashscreen" "/etc/init.d/"
+    cat > "/etc/systemd/system/asplashscreen.service" << _EOF_
+[Unit]
+Description=Show custom splashscreen
+DefaultDependencies=no
+Before=local-fs-pre.target
+Wants=local-fs-pre.target
+ConditionPathExists=$md_inst/asplashscreen.sh
 
-    iniConfig "=" '"' /etc/init.d/asplashscreen
+[Service]
+Type=oneshot
+ExecStart=$md_inst/asplashscreen.sh
+RemainAfterExit=yes
+
+[Install]
+WantedBy=sysinit.target
+_EOF_
+
+    gitPullOrClone "$md_inst" https://github.com/RetroPie/retropie-splashscreens.git
+
+    cp -a "$md_data/asplashscreen.sh" "$md_inst"
+    chown root:root "/etc/systemd/system/asplashscreen.service" "$md_inst/asplashscreen.sh"
+    chmod +x "/etc/systemd/system/asplashscreen.service" "$md_inst/asplashscreen.sh"
```

Comment:
Is the chown needed ? (removing the `-a` should remove the need for the chown, and would still preserve any executable permissions of the .sh script. Also the service doesn't need to be executable agaik.

Reply:
You're right about the executable permissions for the .service file - I'll remove that. As for the chown, it's not necessary, but it could be a security issue to allow a system service to be modified by a user account. I think that git preserves file permissions but not user/group ones.

Reply:
if you dont use `cp -a` the files will be installed as root. cp doesn't copy user ownership by default.